From 557819ffedc3421792c1f41343aad9e617d7237d Mon Sep 17 00:00:00 2001
From: Franck Duriez <franck.lucien.duriez@gmail.com>
Date: Sun, 28 Apr 2024 18:45:58 +0200
Subject: [PATCH 1/3] Make CONFIG_RFTEST=n valid

diff --git a/aic_load_fw/aic_compat_8800d80.c b/aic_load_fw/aic_compat_8800d80.c
index 8f5c56b..93127ef 100644
--- a/aic_load_fw/aic_compat_8800d80.c
+++ b/aic_load_fw/aic_compat_8800d80.c
@@ -415,6 +415,8 @@ int aicfw_download_fw_8800d80(struct aic_usb_dev *usb_dev)
 
     printk("addr_adid 0x%x, addr_patch 0x%x\n", patch_info.addr_adid, patch_info.addr_patch);
 #endif
+
+#ifdef CONFIG_RFTEST
     if(testmode == FW_NORMAL_MODE){
 
         if (chip_id != CHIP_REV_U01){
@@ -671,6 +673,8 @@ int aicfw_download_fw_8800d80(struct aic_usb_dev *usb_dev)
 
             return -1;
         }
+#endif // CONFIG_RFTEST
+
     return 0;
 }
 
diff --git a/aic_load_fw/aicwf_usb.c b/aic_load_fw/aicwf_usb.c
index fd93e83..c8b8a1f 100644
--- a/aic_load_fw/aicwf_usb.c
+++ b/aic_load_fw/aicwf_usb.c
@@ -1379,6 +1379,7 @@ int aicfw_download_fw_8800(struct aic_usb_dev *usb_dev){
 	int i = 0;
     const u32 fw_addr = RAM_FW_ADDR;
     
+#ifdef CONFIG_RFTEST
 #ifdef CONFIG_M2D_OTA_AUTO_SUPPORT
         if(testmode == FW_M2D_OTA_MODE){
             rwnx_plat_m2d_flash_ota_android(usb_dev,FW_M2D_OTA_NAME);
@@ -1588,7 +1589,9 @@ int aicfw_download_fw_8800(struct aic_usb_dev *usb_dev){
             kfree(wakeup_param);
 
             return -1;
-        }else {
+        }else
+#endif // CONFIG_RFTEST
+        {
             if (rwnx_plat_bin_fw_upload_android(usb_dev, RAM_FW_ADDR, FW_BASE_NAME)) {
                 return -1;;
             }
