From db74f59959d3708bc882bc5ab33da93d6dfa73ff Mon Sep 17 00:00:00 2001
From: Franck Duriez <franck.lucien.duriez@gmail.com>
Date: Sun, 28 Apr 2024 18:50:13 +0200
Subject: [PATCH 2/3] Fix DKMS config

diff --git a/dkms.conf b/dkms.conf
index e150abe..92e2b1f 100644
--- a/dkms.conf
+++ b/dkms.conf
@@ -2,10 +2,9 @@ PACKAGE_NAME="aic8800"
 PACKAGE_VERSION="1.0.7"
 BUILT_MODULE_NAME[0]="aic8800_fdrv"
 BUILT_MODULE_LOCATION[0]="aic8800_fdrv/"
-DEST_MODULE_LOCATION[0]="/updates/dkms/"
+DEST_MODULE_LOCATION[0]="/kernel/drivers/net/wireless"
 BUILT_MODULE_NAME[1]="aic_load_fw"
 BUILT_MODULE_LOCATION[1]="aic_load_fw/"
-DEST_MODULE_LOCATION[1]="/updates/dkms/"
-MAKE="arch='$arch' kernelver='$kernelver' ./driverctl make all"
-CLEAN="make clean"
-AUTOINSTALL="YES"
+DEST_MODULE_LOCATION[1]="/kernel/drivers/net/wireless"
+MAKE="make all KSRC=/usr/lib/modules/${kernelver}/build KVER=${kernelver} CONFIG_RWNX_DBG=n CONFIG_RWNX_P2P_DEBUGFS=n CONFIG_AIC_FW_PATH=/usr/lib/firmware CONFIG_RFTEST=n"
+AUTOINSTALL="yes"
